src/main/java/org/broadinstitute/hellbender/tools/spark/sv/CallVariantsFromAlignedContigsSpark.java

```diff
@@ -0,0 +1,440 @@
+        final List<AlignmentRegion> alignmentRegions = IterableUtils.toList(alignmentRegionsIterable);
+        if (alignmentRegions.size() > 1) {
+            alignmentRegions.sort(Comparator.comparing(a -> a.startInAssembledContig));
+        }
+        return getBreakpointAlignmentsFromAlignmentRegions(contigSequence, alignmentRegions, minAlignLength);
+    }
+
+    @VisibleForTesting
+    private static VariantContext filterBreakpointsAndProduceVariants(final Tuple2<BreakpointAllele, Iterable<Tuple2<Tuple2<String, String>, BreakpointAlignment>>> assembledBreakpointsPerAllele, final Broadcast<ReferenceMultiSource> broadcastReference) throws IOException {
+        int numAssembledBreakpoints = 0;
+        int highMqMappings = 0;
+        int midMqMappings = 0;
+        int lowMqMappings = 0;
+        int maxAlignLength = 0;
+
+        final Iterable<Tuple2<Tuple2<String,String>, BreakpointAlignment>> assembledBreakpoints = assembledBreakpointsPerAllele._2;
+        final List<Tuple2<Tuple2<String, String>, BreakpointAlignment>> assembledBreakpointsList = IterableUtils.toList(assembledBreakpoints);
+        final int numBreakpoints = assembledBreakpointsList.size();
+        final List<Integer> mqs = new ArrayList<>(numBreakpoints);
+        final List<Integer> alignLengths = new ArrayList<>(numBreakpoints);
+        final BreakpointAllele breakpointAllele = assembledBreakpointsPerAllele._1;
+        final List<String> breakpointIds = new ArrayList<>(numBreakpoints);
+        final List<String> assembledContigIds = new ArrayList<>(numBreakpoints);
+
+
+        for (final Tuple2<Tuple2<String,String>, BreakpointAlignment> assembledBreakpointPair : assembledBreakpointsList) {
+            final BreakpointAlignment breakpointAlignment = assembledBreakpointPair._2;
+            numAssembledBreakpoints = numAssembledBreakpoints + 1;
+            final int assembledBreakpointMapq = Math.min(breakpointAlignment.region1.mapqual, breakpointAlignment.region2.mapqual);
+            if (assembledBreakpointMapq == 60) {
+                highMqMappings = highMqMappings + 1;
+            } else if (assembledBreakpointMapq > 0) {
+                midMqMappings = midMqMappings + 1;
+            } else {
+                lowMqMappings = lowMqMappings + 1;
+            }
+            mqs.add(assembledBreakpointMapq);
+            final int assembledBreakpointAlignmentLength =
+                    Math.min(breakpointAlignment.region1.referenceInterval.size(),
+                            breakpointAlignment.region2.referenceInterval.size()) - breakpointAlignment.region1.overlapOnContig(breakpointAlignment.region2);
+            alignLengths.add(assembledBreakpointAlignmentLength);
+            maxAlignLength = Math.max(maxAlignLength, assembledBreakpointAlignmentLength);
+            breakpointIds.add(assembledBreakpointPair._1._1);
+            assembledContigIds.add(breakpointAlignment.contigId);
+        }
```

Comment:
I think this `for` block and the related containers could be replaced with something like

```java
        // MQ, alignLen, asmID, contigID
        final List<Tuple4<Integer, Integer, String, String>> breakpointAlignmentInformation = StreamSupport.stream(assembledBreakpointsPerAllele._2.spliterator(), false).map(entry ->{
            final BreakpointAlignment breakpointAlignment = entry._2();
            final int assembledBreakpointAlignmentLength =
                    Math.min(breakpointAlignment.region1.referenceInterval.size(),
                            breakpointAlignment.region2.referenceInterval.size()) - breakpointAlignment.region1.overlapOnContig(breakpointAlignment.region2);
            return new Tuple4<>(Math.min(breakpointAlignment.region1.mapqual, breakpointAlignment.region2.mapqual), // mqs
                    assembledBreakpointAlignmentLength, // alignLengths
                    entry._1._1, //assemblyID
                    entry._1._2);//contigID
        }).collect(Collectors.toList());

        final int maxAlignLength = breakpointAlignmentInformation.stream().map(e -> e._2()).max(Comparator.naturalOrder()).orElse(0);
        final int highMqMappings = (int) breakpointAlignmentInformation.stream().map(e -> e._1()).filter(i -> i==60).count();
```